From 015bdfd1de9135279c7662eb89acc81d1a843202 Mon Sep 17 00:00:00 2001
From: 2155X 2155X <syenergyyy@gmail.com>
Date: Tue, 7 Dec 2021 01:12:35 +0200
Subject: [PATCH] Fix extraction "Dolphin Actions" not abiding "Open
 destination folder after extracting" setting.

BUG: 319134
CCBUG: 298248


(cherry picked from commit 6b3b77e81a05154204ca89140afbec2021ea431b)
---
 app/extractfileitemaction.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/app/extractfileitemaction.cpp b/app/extractfileitemaction.cpp
index b714cee6..cd6b6f96 100644
--- a/app/extractfileitemaction.cpp
+++ b/app/extractfileitemaction.cpp
@@ -14,6 +14,7 @@
 #include <KLocalizedString>
 #include <KPluginFactory>
 
+#include "settings.h"
 #include "mimetypes.h"
 #include "pluginmanager.h"
 #include "batchextract.h"
@@ -96,7 +97,7 @@ QAction *ExtractFileItemAction::createAction(const QIcon& icon, const QString& n
     connect(action, &QAction::triggered, this, [urls,name, option, parent,this]() {
         auto *batchExtractJob = new BatchExtract(parent);
         batchExtractJob->setDestinationFolder(QFileInfo(urls.first().toLocalFile()).path());
-        batchExtractJob->setOpenDestinationAfterExtraction(true);
+        batchExtractJob->setOpenDestinationAfterExtraction(ArkSettings::openDestinationFolderAfterExtraction());
         if (option == AutoSubfolder) {
             batchExtractJob->setAutoSubfolder(true);
         } else if (option == ShowDialog) {
